crates/agglayer-grpc-api/src/node_state_service.rs

```diff
@@ -135,4 +139,201 @@ where
             certificate_header: header.map(Into::into),
         }))
     }
+
+    #[tracing::instrument(level = "debug", skip(self, request), fields(request_id = tracing::field::Empty))]
+    async fn get_network_status(
+        &self,
+        request: tonic::Request<GetNetworkStatusRequest>,
+    ) -> Result<tonic::Response<GetNetworkStatusResponse>, tonic::Status> {
+        let request_id = uuid::Uuid::new_v4().to_string();
+        tracing::Span::current().record("request_id", &request_id);
+        let request = request.into_inner();
+        let network_id = request.network_id.into();
+
+        // Gather all the network information needed to build the response
+
+        // Retrieve latest settled certificate
+        let latest_settled_certificate = self
+            .service
+            .get_latest_settled_certificate_header(network_id)
+            .map_err(|error| {
+                error!(?error, "Failed to get latest settled certificate");
+                tonic::Status::with_error_details(
+                    tonic::Code::NotFound,
+                    "Failed to get latest settled certificate",
+                    ErrorDetails::with_error_info(
+                        GetNetworkStatusErrorKind::MissingLatestSettledCertificate.as_str_name(),
+                        GET_NETWORK_STATUS_METHOD_PATH,
+                        [],
+                    ),
+                )
+            })?;
+
+        let latest_pending_certificate = self
+            .service
+            .get_latest_pending_certificate_header(network_id)
+            .map_err(|error| {
+                error!(?error, "Failed to get latest pending certificate");
+                tonic::Status::with_error_details(
+                    tonic::Code::NotFound,
+                    "Failed to get latest pending certificate",
+                    ErrorDetails::with_error_info(
+                        GetNetworkStatusErrorKind::MissingLatestPendingCertificate.as_str_name(),
+                        GET_NETWORK_STATUS_METHOD_PATH,
+                        [],
+                    ),
+                )
+            })?;
+
+        // Determine network type from the latest available certificate
+        let network_type = match self
+            .service
+            .get_latest_available_certificate_for_network(network_id)
+        {
```

Comment:
could be ok for now but eventually I think that we would expect the type returned by `get_verifier_type` from the smart contracts, like here

https://github.com/agglayer/agglayer/blob/374001dcf239ee80275c0067068a97b61e0e4626/crates/agglayer-aggregator-notifier/src/settlement_client/rpc.rs#L140-L165